core/src/main/java/org/elasticsearch/index/query/FilteredQueryParser.java

```diff
@@ -83,44 +78,35 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             }
         }
 
-        // parsed internally, but returned null during parsing...
         if (query == null) {
-            return null;
+            /*
+             * What the hack happened here? What we got from the user looks something like this:
+             *
+             * {
+             *   "query": {
+             *      "filtered": {
+             *          "query": { },
+             *          "filter": {"range": { "created": { "gte": "now - 1d / d" }}}
+             *      }
+             *   }
+             * }
+             * */
+            return EmptyQueryBuilder.PROTOTYPE;
```

Comment:
this can be null only if the query element wasn't there at all I think, if it was there but empty it would become an EmptyQueryBuilder during `parseContext#parseInnerQueryBuilder`. I will leave it to @cbuescher though to decide whether what we do here is correct :) I am not sure